src/basilisp/io.lpy

```diff
@@ -121,6 +128,18 @@
 
     Callers should generally prefer :lpy:fn:`output-stream` to this function."))
 
+(defn- convert-to-path-or-url [f-str]
+  "Convert ``f-str`` to a python Path or URL object based on to
+  whether it represents an absolute path or not, respectively.
+
+This fn is intended to be used with the input and output
+writers. Converting MS-Windows absolutely paths (such as c:\\xyz and
+\\\\share\\xyz) directly to URLs are likely to confuse the urllib
+parser. As such, they are converted to Path objects instead."
+  (if (os.path/isabs f-str)
+          (pathlib/Path f-str)
+          (urllib.parse/urlparse f-str)))
```

Comment:
```suggestion
  "Convert ``f-str`` to a python Path or URL object based on to
  whether it represents an absolute path or not, respectively.

  This fn is intended to be used with the input and output
  writers. Converting MS-Windows absolutely paths (such as c:\\xyz and
  \\\\share\\xyz) directly to URLs are likely to confuse the urllib
  parser. As such, they are converted to Path objects instead."
  (if (os.path/isabs f-str)
    (pathlib/Path f-str)
    (urllib.parse/urlparse f-str)))
```

Reply:
Done

Reply:
I think this might've gotten reverted in the rebase.